# 🛒 Projeto: Análise de Vendas e Lucratividade em E-commerce

## 🎯 Objetivo
Explorar o desempenho de vendas por canal, margem de lucro e comportamento de produtos, com foco em canais como Shiprocket, INCREFF, Amazon, etc.

## 📂 Dataset
Arquivo: `Sale Report.csv`

## 🔍 Etapa 1: Carregamento e Diagnóstico Inicial

- Ver estrutura (`.info()`)
- Ver primeiras linhas (`.head()`)
- Ver estatísticas (`.describe()`)

## 🧾 Observações Iniciais

- Colunas identificadas:
  - SKU
  - Date of Sale
  - Amount Paid
  - TP1, TP2
  - Shiprocket, INCREFF
- Há colunas com valores ausentes? (verificar via `.isnull()`)

## 📌 Próximas Etapas

- Criar dicionário de colunas
- Analisar por canal (Shiprocket vs INCREFF)
- Calcular margens por produto
- Visualizar evolução temporal


In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('../data/Sale Report.csv')

In [3]:
df

index       SKU Code Design No.  Stock       Category Size Color
0         0    AN201-RED-L      AN201    5.0  AN : LEGGINGS    L   Red
1         1    AN201-RED-M      AN201    5.0  AN : LEGGINGS    M   Red
2         2    AN201-RED-S      AN201    3.0  AN : LEGGINGS    S   Red
3         3   AN201-RED-XL      AN201    6.0  AN : LEGGINGS   XL   Red
4         4  AN201-RED-XXL      AN201    3.0  AN : LEGGINGS  XXL   Red
...     ...            ...        ...    ...            ...  ...   ...
9266   9266            NaN        NaN    NaN            NaN  NaN   NaN
9267   9267            NaN        NaN    NaN            NaN  NaN   NaN
9268   9268            NaN        NaN    NaN            NaN  NaN   NaN
9269   9269            NaN        NaN    NaN            NaN  NaN   NaN
9270   9270            NaN        NaN    NaN            NaN  NaN   NaN

[9271 rows x 7 columns]

In [4]:
df.head()

index       SKU Code Design No.  Stock       Category Size Color
0      0    AN201-RED-L      AN201    5.0  AN : LEGGINGS    L   Red
1      1    AN201-RED-M      AN201    5.0  AN : LEGGINGS    M   Red
2      2    AN201-RED-S      AN201    3.0  AN : LEGGINGS    S   Red
3      3   AN201-RED-XL      AN201    6.0  AN : LEGGINGS   XL   Red
4      4  AN201-RED-XXL      AN201    3.0  AN : LEGGINGS  XXL   Red

In [5]:
df.tail()

index SKU Code Design No.  Stock Category Size Color
9266   9266      NaN        NaN    NaN      NaN  NaN   NaN
9267   9267      NaN        NaN    NaN      NaN  NaN   NaN
9268   9268      NaN        NaN    NaN      NaN  NaN   NaN
9269   9269      NaN        NaN    NaN      NaN  NaN   NaN
9270   9270      NaN        NaN    NaN      NaN  NaN   NaN

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9271 entries, 0 to 9270
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   index       9271 non-null   int64  
 1   SKU Code    9188 non-null   object 
 2   Design No.  9235 non-null   object 
 3   Stock       9235 non-null   float64
 4   Category    9226 non-null   object 
 5   Size        9235 non-null   object 
 6   Color       9226 non-null   object 
dtypes: float64(1), int64(1), object(5)
memory usage: 507.1+ KB


In [7]:
df.describe()

index        Stock
count  9271.000000  9235.000000
mean   4635.000000    26.246454
std    2676.451507    58.462891
min       0.000000     0.000000
25%    2317.500000     3.000000
50%    4635.000000     8.000000
75%    6952.500000    31.000000
max    9270.000000  1234.000000

## 🔎 Diagnóstico Técnico Inicial

- O dataset possui 9271 linhas e 7 colunas.
- Nenhuma coluna, exceto o índice, está 100% preenchida.
- As colunas `SKU Code`, `Design No.`, `Category`, `Size` e `Color` são todas do tipo `object` (texto).
- A coluna `Stock` é numérica e apresenta grande variação:
  - Produtos com estoque **zerado**
  - Alguns com mais de **1000 unidades**
  - A maioria com **estoques baixos** (mediana = 8)
- Será importante verificar a relação entre estoque e vendas/margem mais tarde.
- 25% dos produtos têm estoque de até 3 unidades.
- 50% dos produtos têm até 8 unidades no estoque
- 75% dos produtos tem até 31 unidades.
- Possível outliner puxando a média para cima (max = 1234)
